index.js

```diff
@@ -51,7 +59,152 @@ module.exports = function (app) {
     return new Date(new Date().getTime() - 60 * 60 * 1000).toISOString()
   }
 
-  return {
+  function getQuery(startTime, endTime, path) {
+    let skPath
+    let contextClause = ''
+
+    if ( path != null ) {
+      if ( path.length == 1 ) {
+        contextClause = `and context =~ /${path[0]}.*/`
+      } else if ( path.length > 1 ) {
+        let context = path[1]
+        if ( context == 'self' ) {
+          context = app.selfId
+        }
+        contextClause = `and context =~ /${path[0]}\.${context}\..*/`
+      }
+      
+      skPath = path.slice(2).join('.')
+      if ( skPath.length === 0 ) {
+        skPath = null
+      }
+    }
+    
+    const measurements = skPath ? `/${skPath}.*/` : '/.*/'
+    let whereClause = `time > '${startTime.toISOString()}' and time <= '${endTime.toISOString()}' ${contextClause}`
```

Comment:
const and let are not as useful as they could be, but let's not mix them for no reason.